snuba/util.py

```diff
@@ -34,9 +35,19 @@
 
 T = TypeVar("T")
 
+# example partition name: "(90, '2018-03-13 00:00:00')"
+ERRORS_TRANSACTIONS_PART_RE = re.compile(
+    r"\((?P<retention>\d+),\s*('(?P<timestamp>\d{4}-\d{2}-\d{2})')\)"
+)
+PART_RE = {
+    StorageKey.EVENTS: re.compile(
+        r"\('(?P<timestamp>\d{4}-\d{2}-\d{2})',\s*(?P<retention>\d+)\)"
+    ),
+    StorageKey.ERRORS: ERRORS_TRANSACTIONS_PART_RE,
+    StorageKey.TRANSACTIONS: ERRORS_TRANSACTIONS_PART_RE,
+}
+
```

Comment:
I don't think having partition schemas defined in snuba util for all tthe storages in the data model is a great idea. It seems coupled with the migration that defines the schema but it is kept in a place that is separate enough to be forgotten.
Since the partition key is something we need to use outside the the migration framework it should be defined as a property of a storage (just like its columns set). 

This does not sound great either. Can we generate it from the database schema instead from the system.tables table in a similar way as we do for comparing the DB schema with the expected schema ?

Reply:
That would be ideal, but I don't think we can easily derive the meaning about what each component of the partition key represents or when partitions can be cleaned up just from knowing what the partition key on a particular table is (although yes that can be looked up).

So the options I considered were just mapping the key format to the storage here, or making it a property of the storage itself. Neither is great, which is why I was hesitant to make it a bona fide property of the storage.